controllers/validation/validation.go

```diff
@@ -102,9 +103,14 @@ func validateArbiterSpec(mdb mdbv1.MongoDBCommunity) error {
 }
 
 // validateAuthModeSpec checks that the list of modes does not contain duplicates.
-func validateAuthModeSpec(mdb mdbv1.MongoDBCommunity) error {
+func validateAuthModeSpec(mdb mdbv1.MongoDBCommunity, log *zap.SugaredLogger) error {
 	allModes := mdb.Spec.Security.Authentication.Modes
 
+	// Issue warning if Modes array is empty
+	if len(allModes) == 0 {
+		log.Warnf("Modes array is empty")
```

Comment:
The warning message should be a bit more descriptive and should provide at least some level of guidance to the user. Here's an example: 

> An empty Modes array has been provided. The default mode (SCRAM-SHA-256) will be used.

Reply:
Updated